src/store/Reducers/auth.js

```diff
@@ -0,0 +1,37 @@
+import actions from '../Actions/typeActions'
+
+const initialState = {
+  email: 'email',
+  password: '',
+  permissions: {},
+  token: null,
+  username: 'userName',
+  userId: null,
+  validTonken: false,
+}
+
+const auth = (state = initialState, action) => {
+  switch (action.type) {
+    case actions.AUTH.LOGIN:
+      let auth = {
+        ...state,
+      }
+      if (action.payload.token) {
+        auth = {
+          ...auth,
+          ...action.payload,
+        }
+      }
+
+      return auth
```

Comment:
deveria ser assim essa alteração no state
```suggestion
       return ({
          ...state,
          ...action.payload,
        })
```